packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -320,6 +324,7 @@ export async function scanParentDirs(
   } else if (hasYarnLock) {
     cliType = 'yarn';
     lockfilePath = yarnLockPath;
+    lockfileVersion = Number(yarnLockYaml.__metadata.version);
```

Comment:
I see this field in yarn3 and yarn4, but need to check yarn 1 and 2 also

Reply:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (blocking)** ‏

I think this needs to be careful when accessing `yarnLockYaml.__metadata`. I don't see that in a `yarn@1` lockfile, for example.

We could look for the comment:

```
# yarn lockfile v1
```

and parse it out, but I don't know if that version changes in `yarn@1` at all. Maybe we can assume lockfile v1?

```suggestion
    lockfileVersion = Number(yarnLockYaml.__metadata?.version) || 1;
```

---

**Yarn 1**

```
$ yarn -v
1.22.19

$ head yarn.lock
# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


"@ampproject/remapping@^2.2.0":
  version "2.2.0"
  resolved "https://registry.yarnpkg.com/@ampproject/remapping/-/remapping-2.2.0.tgz#56c133824780de3174aed5ab6834f3026790154d"
  integrity sha512-qRmjj8nj9qmLTQXXmaR1cck3UXSRMPrbsLJAasZpF+t3riI71BXed5ebIOYwQntykeZuhjsdweEc9BxH5Jc26w==
  dependencies:
    "@jridgewell/gen-mapping" "^0.1.0"
```

**Yarn 2**

```
$ yarn -v
2.4.3

$ head yarn.lock
# This file is generated by running "yarn install" inside your project.
# Manual changes might be lost - proceed with caution!

__metadata:
  version: 4
  cacheKey: 7

"@planetscale/database@npm:^1.0.1":
  version: 1.11.0
  resolution: "@planetscale/database@npm:1.11.0"
```